src/sb_utils.R

```diff
@@ -8,25 +24,100 @@ sb_replace_files <- function(sb_id, ..., file_hash){
   hashed_filenames <- c()
   if (!missing(file_hash)){
     hashed_filenames <- yaml.load_file(file_hash) %>% names %>% sort() %>% rev()
-    for (file in hashed_filenames){
-      item_replace_files(sb_id, files = file)
+    if(use_task_table) {
+      do_item_replace_tasks(sb_id, hashed_filenames, sources)
+    } else {
+      for (file in hashed_filenames){
+        item_replace_files(sb_id, files = file)
+      }
     }
   }
-  files <- c(...)
+  
   if (length(files) > 0){
-    item_replace_files(sb_id, files = files)
+    if(use_task_table) {
+      do_item_replace_tasks(sb_id, files, sources)
+    } else {
+      item_replace_files(sb_id, files = files)
+    }
   }
   
 }
 
-sb_render_post_xml <- function(sb_id, ..., xml_file = NULL){
+sb_render_post_xml <- function(sb_id, ..., xml_file = NULL, use_task_table = TRUE, sources){
   
   if (is.null(xml_file)){
     xml_file <- file.path(tempdir(), paste0(sb_id,'.xml'))
   }
   
   render(filename = xml_file, ...)
   
-  sb_replace_files(sb_id = sb_id, xml_file)
+  sb_replace_files(sb_id = sb_id, xml_file, use_task_table, sources)
+  
+}
+
+# Helper function to create a task_table for the files that need to be pushed to SB
+do_item_replace_tasks <- function(sb_id, files, sources) {
+  
+  # Define task table columns
+  sb_push <- scipiper::create_task_step(
+    step_name = 'push_file_to_sb',
+    target_name = function(task_name, step_name, ...){
```

Comment:
Your target name makes it look like this is a file instead of an object, which causes a fail. 

I don't see anything wrong with using a simple generic naming scheme, such as `uploaded_file_01` where your `tasks` are 1:length(files) and you select the file name for `upload_and_record()` from the index instead of using the task_name. 

something like

```r
task_df <- tibble(filepath = files) %>% mutate(task_name = paste0('uploaded_file_', row_number())
```
Then filtering within the command for the push to select the appropriate file name as an input